src/libraries/System.Private.CoreLib/src/System/Index.cs

```diff
@@ -15,12 +15,7 @@ namespace System
     /// int lastElement = someArray[^1]; // lastElement = 5
     /// </code>
     /// </remarks>
-#if SYSTEM_PRIVATE_CORELIB
-    public
-#else
-    internal
-#endif
-    readonly struct Index : IEquatable<Index>
+    public readonly struct Index : IEquatable<Index>
```

Comment:
This should be deleted, too:
https://github.com/dotnet/runtime/blob/99e94ce55d79ea7d702f4d1e0c5fc70f2677670f/src/libraries/Directory.Build.targets#L205

Reply:
This will be done when we address https://github.com/dotnet/runtime/issues/104308.